ui/src/controller/track_decider.ts

```diff
@@ -422,11 +421,13 @@ class TrackDecider {
         (track.trackGroup && track.trackGroup !== SCROLLING_TRACK_GROUP)) {
         continue;
       }
-      if (MEMORY_USAGE_COUNTERS.includes(track.name)) {
+      // Check if track is a GPU counter
+      const result = await this.engine.query('select description from gpu_counter_track where name = "' + track.name + '"');
```

Comment:
Instead of issuing a query for each track, we might want to make one query to get the names (preferably IDs because track names can repeat) of the tracks that are GPU counter tracks, then correlate those result rows with the tracks created by the `TrackDecider`.

Reply:
Are we safe to assume that the table has an Id set for each track and that the AddTrackArgs also has an Id set?

Reply:
Every track has an unique ID, assigned sequentially, yes. And if the `config` object for a GPU track doesn't include the track ID, we could always just add it.

Reply:
So, you'd wanna see something like this instead?
```
const counterTrackIds: string[] = Array.from((await this.engine.query(`select id from gpu_counter_track`)).iter({id:NUM}));
    for (const track of this.tracksToAdd) {
      if (track.kind !== COUNTER_TRACK_KIND ||
        (track.trackGroup && track.trackGroup !== SCROLLING_TRACK_GROUP)) {
        continue;
      }
      if (track.id && counterTrackIds.includes(track.id)) {
        track.trackGroup = this.lazyTrackGroup('GPU Counters', {collapsed: false, parentGroup: this.gpuGroup()})();
      } else {
        track.trackGroup = this.lazyTrackGroup('Memory Usage')();
      }
    }
    ```

Reply:
Yes, something like that. 😀  Although there's a mismatch between the query's definition of the `id` column type as `NUM` and the `counterTrackIds` being an array of `string`.

Reply:
Array.from() didn't work with the typing.  And when I tried using the counter Ids it didn't seem to equate the one from the table to the one associated with the track, so I used names since Ids at least according to AddTrackArgs can be undefined while names can't.